src/decimal128.rs

```diff
@@ -469,6 +469,16 @@ impl std::str::FromStr for ParsedDecimal128 {
 }
 
 fn round_decimal_str(s: &str, precision: usize) -> Result<&str, ParseError> {
+    // TODO: In 1.80+ there's split_at_checked to make sure the split doesn't
+    // panic if the index doesn't falls at a codepoint boundary, until then
+    // we can check it with s.is_char_boundary(precision)
+    if !s.is_char_boundary(precision) {
+        // a non-digit (all single byte utf8) would trigger a ParseIntError::InvalidDigit,
+        // so here we generate a ParseIntError::InvalidDigit kind of error too.
+        return Err(ParseError::InvalidCoefficient(
```

Comment:
I think I'd rather have a new enum variant for `ParseError` (something like `ParseError::Unparseable`) than synthesize an error this way; that way it could also carry the source string to make debugging easier.

Reply:
I'm not saying I disagree, but no other error variants carry more information than this one.

`InvalidDigit` is arguably correct, as that would be the error if the split was character-based (instead of byte-based). The only reason I didn't do that is that it's more and slower code (linear vs. constant time).

Reply:
If we could directly construct an `InvalidDigit` I'd have less concern.  My worry is that by synthesizing it this way it'll only _probably_ continue to be `InvalidDigit`; the internals of `ParseIntError` are private so there are no guarantees of stability.  If the error changes in the future to include the offending text, for example, it'll be very confusing for downstream users of this crate to see `"❤"` that doesn't actually appear in their data.

Adding a new variant to `ParseError` avoids that issue, and if we're doing that it may as well carry the text.

Reply:
I'll add a variant. But also, I'd like to point out that InvalidDigit (and any variant of ParseIntError for that matter) do _not_ carry information about the input, such as the offending character.